File: snipxn-backend/snipxn-app/src/main/java/com/snipxn/feedback/entity/Feedback.java
package com.snipxn.feedback.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

import java.time.OffsetDateTime;

@Getter
@Setter
@TableName("feedbacks")
public class Feedback {

    @TableId(type = IdType.ASSIGN_UUID)
    private String id;

    private String userId;

    private String content;

    private String contact;

    private String images;

    private String status;

    private String adminReply;

    @TableField(fill = FieldFill.INSERT)
    private OffsetDateTime createdAt;
}

File: snipxn_app/src/api/feedback.ts
import { apiClient } from './axios';

export interface SubmitFeedbackRequest {
  content: string;
  contact: string | null;
  images: string[];
}

export interface SubmitFeedbackResponse {
  id: string;
  status: string;
  createdAt: string | null;
  confirmationEmailQueued: boolean;
}

export function submitFeedback(
  payload: SubmitFeedbackRequest,
): Promise<SubmitFeedbackResponse> {
  return apiClient.post<SubmitFeedbackResponse, SubmitFeedbackRequest>(
    '/feedbacks',
    payload,
  );
}
